From 263348466169e029090479079c97663ddc94cb6c Mon Sep 17 00:00:00 2001
From: Thomas Perl <thomas.perl@jolla.com>
Date: Tue, 22 Oct 2013 12:15:11 +0200
Subject: [PATCH 20/65] zypp: Force failure for missing ssu.ini

In cases of a missing ssu.ini, the do { ... } while (retry) loop in
zypp/media/MediaCurl.cc does not exit, even if MediaCurl::authenticate()
(and in turn our implementation of prompt()) returns false (=abort).

To fix this, we now throw a libzypp zypp::media::MediaException in the
prompt() function in order to force the backend and libzypp to give up.
---
 backends/zypp/pk-backend-zypp.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index 9b1b9d7da..417b06a4b 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -83,6 +83,7 @@
 #include <zypp/base/Logger.h>
 #include <zypp/base/String.h>
 #include <zypp/media/MediaManager.h>
+#include <zypp/media/MediaException.h>
 #include <zypp/parser/IniDict.h>
 #include <zypp/parser/ParseException.h>
 #include <zypp/parser/ProductFileReader.h>
@@ -447,7 +448,8 @@ struct AuthenticationReportReceiver : public zypp::callback::ReceiveReport<zypp:
 		std::string urlstr = url.asString();
 		PK_ZYPP_LOG("Needs authentication: %s", urlstr.c_str());
 		/* No interactive authentication supported - admit failure */
-		return false;
+		ZYPP_THROW(zypp::media::MediaException("Authentication failed (is SSU set up correctly?)"));
+		return false; // Not reached
 	}
 };
 
-- 
2.13.5

